internal/provider/vcs_resource.go

```diff
@@ -0,0 +1,505 @@
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *VcsResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state VcsResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	vcsRequest, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/api/v1/organization/%s/vcs/%s", r.endpoint, state.OrganizationId.ValueString(), state.ID.ValueString()), nil)
+	vcsRequest.Header.Add("Authorization", fmt.Sprintf("Bearer %s", r.token))
+	vcsRequest.Header.Add("Content-Type", "application/vnd.api+json")
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating VCS resource request", fmt.Sprintf("Error creating VCS resource request: %s", err))
+		return
+	}
+
+	vcsResponse, err := r.client.Do(vcsRequest)
+	if err != nil {
+		resp.Diagnostics.AddError("Error executing VCS resource request", fmt.Sprintf("Error executing VCS resource request: %s", err))
+		return
+	}
+
+	bodyResponse, err := io.ReadAll(vcsResponse.Body)
+	if err != nil {
+		tflog.Error(ctx, fmt.Sprintf("Error reading organization variable resource response, error: %s, response status: %s", err, vcsResponse.Status))
+	}
+	vcs := &client.VcsEntity{}
+
+	tflog.Info(ctx, "Body Response", map[string]any{"bodyResponse": string(bodyResponse)})
+	err = jsonapi.UnmarshalPayload(strings.NewReader(string(bodyResponse)), vcs)
+
+	if err != nil {
+		resp.Diagnostics.AddError("Error unmarshal payload response", fmt.Sprintf("Error unmarshal payload response, error: %s, response status: %s", err, vcsResponse.Status))
+		return
+	}
+
+	tflog.Info(ctx, "Body Response", map[string]any{"bodyResponse": string(bodyResponse)})
+
+	state.ID = types.StringValue(vcs.ID)
+	state.Name = types.StringValue(vcs.Name)
+	state.Description = types.StringValue(vcs.Description)
+	state.VcsType = types.StringValue(vcs.VcsType)
+	state.ClientId = types.StringValue(vcs.ClientId)
+	state.Endpoint = types.StringValue(vcs.Endpoint)
+	state.ApiUrl = types.StringValue(vcs.ApiUrl)
+	state.Status = types.StringValue(vcs.Status)
+
+	// Set refreshed state
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if vcs.Status == "PENDING" {
+		tflog.Warn(ctx, fmt.Sprintf("VCS connection is pending, please logon to %s to connect. Check doc here %s", state.ConnectUrl, helpers.GetVCSProviderDoc()))
+	}
+
+	tflog.Info(ctx, "VCS Resource read succeed", map[string]any{"success": true})
+}
+
+func (r *VcsResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Retrieve values from plan
+	var plan VcsResourceModel
+	var state VcsResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	bodyRequest := &client.VcsEntity{
+		Name:         plan.Name.ValueString(),
```

Comment:
yes updated. Thanks. 

haha, can't believe it's so obvious. I remember previously it worked, but don't know how that part got missed/deleted. 

I blame vim plugin in vscode instead of my intelligence.